interfaces/builtin/content_test.go

```diff
@@ -376,3 +395,209 @@ slots:
+	mountSpec := &mount.Specification{}
+	c.Assert(mountSpec.AddConnectedPlug(s.iface, plug, nil, slot, nil), IsNil)
+	apparmorSpec := &apparmor.Specification{}
+	c.Assert(apparmorSpec.AddConnectedPlug(s.iface, plug, nil, slot, nil), IsNil)
+
+	// Analyze the mount specification.
+	expectedMnt := []mount.Entry{{
+		Name:    "/var/snap/producer/common/read-common",
+		Dir:     "/var/snap/consumer/common/import/read-common",
+		Options: []string{"bind", "ro"},
+	}, {
+		Name:    "/var/snap/producer/2/read-data",
+		Dir:     "/var/snap/consumer/common/import/read-data",
+		Options: []string{"bind", "ro"},
+	}, {
+		Name:    "/snap/producer/2/read-snap",
+		Dir:     "/var/snap/consumer/common/import/read-snap",
+		Options: []string{"bind", "ro"},
+	}, {
+		Name:    "/var/snap/producer/common/write-common",
+		Dir:     "/var/snap/consumer/common/import/write-common",
+		Options: []string{"bind"},
+	}, {
+		Name:    "/var/snap/producer/2/write-data",
+		Dir:     "/var/snap/consumer/common/import/write-data",
+		Options: []string{"bind"},
+	}}
+	c.Assert(mountSpec.MountEntries(), DeepEquals, expectedMnt)
+
+	// Analyze the apparmor specification.
+	c.Assert(apparmorSpec.SecurityTags(), DeepEquals, []string{"snap.consumer.app"})
+	expected := `
+# In addition to the bind mount, add any AppArmor rules so that
+# snaps may directly access the slot implementation's files. Due
+# to a limitation in the kernel's LSM hooks for AF_UNIX, these
+# are needed for using named sockets within the exported
+# directory.
+/var/snap/producer/common/write-common/** mrwklix,
+/var/snap/producer/2/write-data/** mrwklix,
+
+# In addition to the bind mount, add any AppArmor rules so that
+# snaps may directly access the slot implementation's files
+# read-only.
+/var/snap/producer/common/read-common/** mrkix,
+/var/snap/producer/2/read-data/** mrkix,
+/snap/producer/2/read-snap/** mrkix,
+`
+	c.Assert(apparmorSpec.SnippetForTag("snap.consumer.app"), Equals, expected)
+}
+
+func (s *ContentSuite) TestModernContentInterfacePlugins(c *C) {
+	// Define one app snap and two snaps plugin snaps.
+	plug := MockPlug(c, `name: app
+plugs:
+ plugins:
+  interface: content
+  content: plugin-for-app
+  target: $SNAP/plugins
+apps:
+ app:
+  command: foo
+
+`, &snap.SideInfo{Revision: snap.R(1)}, "plugins")
+
+	// XXX: realistically the plugin may be a single file and we don't support
+	// those very well.
+	slotOne := MockSlot(c, `name: plugin-one
+slots:
+ plugin-for-app:
+  interface: content
+  source:
+    read: [$SNAP/plugin]
+`, &snap.SideInfo{Revision: snap.R(1)}, "plugin-for-app")
+
+	slotTwo := MockSlot(c, `name: plugin-two
+slots:
+ plugin-for-app:
+  interface: content
+  source:
+    read: [$SNAP/plugin]
+`, &snap.SideInfo{Revision: snap.R(1)}, "plugin-for-app")
+
+	// Create the mount and apparmor specifications.
+	mountSpec := &mount.Specification{}
+	apparmorSpec := &apparmor.Specification{}
+	for _, slot := range []*interfaces.Slot{slotOne, slotTwo} {
```

Comment:
I understand that the order in which the slots are connected will influence which of the snaps gets mounted at `plugin-2`? Once the connection is made, is the order preserved across reboots?

Reply:
The connection reload logic is stable so it should, yes.